truss/templates/server/common/truss_server.py

```diff
@@ -1,248 +1,169 @@
+    _endpoints: BasetenEndpoints
+    _model: ModelWrapper
 
-        IOLoop.current().start()
+    def __init__(self, http_port: int, model: ModelWrapper):
+        super().__init__(
+            http_port=http_port,
+            enable_grpc=False,
+            workers=1,
+            enable_docs_url=False,
+            enable_latency_logging=False,
+        )
+
+        self._model = model
+        self._endpoints = BasetenEndpoints(
+            self.dataplane, self.model_repository_extension
+        )
+
+    def start_model(self) -> None:
+        """
+        Overloaded version of super().start to use instance model in TrussServer
+        """
+        super().start([self._model])
 
     def create_application(self):
-        return tornado.web.Application(
-            [
-                # Server Liveness API returns 200 if server is alive.
-                (r"/", LivenessHandler),
-                (r"/v1/models", ListHandler, dict(models=self.registered_models)),
-                # Model Health API returns 200 if model is ready to serve.
-                (
-                    r"/v1/models/([a-zA-Z0-9_-]+)",
-                    HealthHandler,
-                    dict(models=self.registered_models),
-                ),
-                (
-                    r"/v1/models/([a-zA-Z0-9_-]+):predict",
-                    PredictHandler,
-                    dict(models=self.registered_models),
+        return FastAPI(
+            title="Baseten Inference Server",
+            docs_url=None,
+            redoc_url=None,
+            default_response_class=ORJSONResponse,
+            on_startup=[self._model.start_load],
+            routes=[
+                # liveness endpoint
+                FastAPIRoute(r"/", self.dataplane.live),
+                # readiness endpoint
+                FastAPIRoute(
+                    r"/v1/models/{model_name}", self._endpoints.model_ready, tags=["V1"]
                 ),
-                (
-                    r"/v1/models/([a-zA-Z0-9_-]+):predict_binary",
-                    TrussPredictHandler,
-                    dict(models=self.registered_models),
+                FastAPIRoute(
+                    r"/v1/models/{model_name}:predict",
+                    self._endpoints.predict,
+                    methods=["POST"],
+                    tags=["V1"],
                 ),
-                (
-                    r"/v1/models/([a-zA-Z0-9_-]+):explain",
-                    ExplainHandler,
-                    dict(models=self.registered_models),
+                FastAPIRoute(
+                    r"/v1/models/{model_name}:predict_binary",
+                    self._endpoints.predict,
+                    methods=["POST"],
+                    tags=["V1"],
                 ),
-                (
-                    r"/v1/models/([a-zA-Z0-9_-]+):explain_binary",
-                    TrussExplainHandler,
-                    dict(models=self.registered_models),
+                FastAPIRoute(
```

Comment:
was `explain_binary` dropped intentionally here? seems deprecated, but just curious why we kept `explain` but not `explain_binary`

Reply:
Yeah, I call that out in the description but I don't think we use it anywhere. I looked for occurrences in the baseten code but it doesn't show up anywhere (that I saw).   The json explain is still there if customers want to use, but we don't need the binary version unless we use it in the baseten app.